clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/LeaderProducerCallback.java

```diff
@@ -101,12 +101,16 @@ public void onCompletion(RecordMetadata recordMetadata, Exception e) {
       // followers consuming from VTs. So it is safe to skip adding the record to leader's StorageBufferService
       if (partitionConsumptionState.getLeaderFollowerState() == LEADER
           && recordMetadata.partition() != partitionConsumptionState.getPartition()) {
+        partitionConsumptionState.mayRemoveTransientRecord(
```

Comment:
I don't think this is correct since the transientRecordCache shouldn't be cleared until the produced message is persisted in local RocksDB instance, otherwise the later lookup from RocksDB will return stale result.